dds/DCPS/transport/framework/NetworkAddress.h

```diff
@@ -86,9 +86,15 @@ bool set_socket_multicast_ttl(const ACE_SOCK_Dgram& socket, const unsigned char&
 extern OpenDDS_Dcps_Export
 bool open_appropriate_socket_type(ACE_SOCK_Dgram& socket, const ACE_INET_Addr& local_address, int* proto_family = 0);
 
+extern OpenDDS_Dcps_Export
+ACE_INET_Addr choose_single_coherent_address(const OPENDDS_VECTOR(ACE_INET_Addr)& addrs, bool prefer_loopback = true, const std::string& name = std::string());
```

Comment:
`name` should be allocator-friendly -- either just `const char*` or the typedef that supports allocators